apps/prairielearn/src/components/TanstackTable.tsx

```diff
@@ -88,19 +81,47 @@ function ResizeHandle({
   );
 }
 
-export function StudentsTable({
+const DefaultEmptyState = (
+  <>
+    <i class="bi bi-search display-4 mb-2" aria-hidden="true" />
+    <p class="mb-0">No results found matching your search criteria.</p>
+  </>
+);
+
+interface TanstackTableProps<RowDataModel> {
+  table: Table<RowDataModel>;
+  filters?: Record<string, (props: { header: Header<RowDataModel, unknown> }) => JSX.Element>;
+  rowHeight?: number;
+  emptyState?: JSX.Element;
+}
+
+const DEFAULT_FILTER_MAP = {};
+
+/**
+ * A generic component that renders a full-width, resizeable Tanstack Table.
+ * @param params
+ * @param params.table - The table model
+ * @param params.title - The title of the table
+ * @param params.filters - The filters for the table
+ * @param params.rowHeight - The height of the rows in the table
+ * @param params.emptyState - The empty state for the table
+ */
+export function TanstackTable<RowDataModel>({
```

Comment:
The reviewer should look at this implementation to ensure it is flexible, but not _too_ flexible (i.e. allows changing stuff we don't care about).